### DISCUSSION TITLE
Entitlement com.apple.security.automation.apple-events

### DISCUSSION THREAD

**Discussion by flemingjoseph** [sparkle-project/Sparkle#2320] [Q&A]
macOS is prompting (non-fatal error) to add Entitlement com.apple.security.automation.apple-events. Is this needed for Sparkle framework to work ?

macOS: 12.6 Intel
Sparkle: v2.0.0

Prompting policy for hardened runtime; service: kTCCServiceAppleEvents requires entitlement com.apple.security.automation.apple-events but it is missing for accessing={<TCCDProcess: identifier=org.sparkle-project.Sparkle.Updater, pid=26566, auid=501, euid=501, binary_path=/Users/fleming/Library/Caches/com.x.y.z/org.sparkle-project.Sparkle/Launcher/Mb0wiBt0u/Updater.app/Contents/MacOS/Updater>}, requesting={<TCCDProcess: identifier=com.apple.appleeventsd, pid=442, auid=55, euid=55, binary_path=/System/Library/CoreServices/appleeventsd>},

**Comment by zorgiepoo** [comment]
Evidently no, this hasn't made updating not work for anyone.

Chances are this won't happen during real deployment if your app is code signed / notarized properly since Sparkle's Updater helper and your own app are signed with the same team ID:

https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_security_automation_apple-events?language=objc

> Your app doesn’t need the Apple Events Entitlement if it only sends Apple events to itself or to other processes signed with the same team ID.
